### DISCUSSION TITLE
Snapshot version scheme recommendation

### DISCUSSION THREAD

**Discussion by lefou** [coursier/coursier#2136] [Q&A] [answered]
Coursier uses the following version pattern for snapshot releases: `major.minor.patch+commitcount-hash`. 
Is there any benefit over the one used by other projects, e.g. mill: `major.minor.patch-commitcount-hash`?
Both differ in the `+` symbol vs. the `-` symbol between version and version suffix. Does the `+` symbol behave better in version ordering?
E.g. in Maven the minus symbol between the version and the rest results in ordering the snapshot release before the patch release, whereas they are released later. Is the plus symbol handled differently and thus the better option?

**Comment by eed3si9n** [comment, marked as answer]
For artifacts following Semantic Versioning, `+` means build metadata (https://semver.org/#spec-item-10) and:
1. It must appear _after_ `-<pre-release>`, and
2. It must be ignored when determining version precedence

so `major.minor.patch+commitcount-hash` does not conform to Semantic Versioning scheme.

I'd suggest something like `<major>.<minor>.<patch>-alpha<commitcount>+<hash>` instead. This would look like `1.0.1-alpha10+1234abcd`. Hopefully `-alpha10` would be recognized universally as unstable. The only caveat is that it would sort before the supposedly tagged `1.0.1`.

**Reply by lefou** [reply]
Ok, thanks for clarifying! To conclude: No, the `+` instead of the `-` is not an improvement for Maven/Ether. It's even invalid  scheme for SemVer.
